pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -228,6 +228,40 @@ public void initialize(ServiceConfiguration conf, PulsarResources pulsarResource
         return allowTheSpecifiedActionOpsAsync(namespaceName, role, authenticationData, AuthAction.sinks);
     }
 
+    private CompletableFuture<Boolean> allowConsumeOrProduceOpsAsync(NamespaceName namespaceName,
+                                                                     String role,
+                                                                     AuthenticationDataSource authenticationData) {
+        CompletableFuture<Boolean> finalResult = new CompletableFuture<>();
+        allowTheSpecifiedActionOpsAsync(namespaceName, role, authenticationData, AuthAction.produce)
+                .whenComplete((produceAuthorized, e) -> {
+                    if (e == null) {
+                        if (produceAuthorized) {
+                            finalResult.complete(produceAuthorized);
+                            return;
+                        }
+                    } else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Namespace [{}] Role [{}] exception occurred while trying to check Produce "
+                                    + "permissions. {}", namespaceName, role, e.getMessage());
+                        }
+                    }
+                    allowTheSpecifiedActionOpsAsync(namespaceName, role, authenticationData, AuthAction.consume)
+                            .whenComplete((consumeAuthorized, ex) -> {
+                                if (ex == null) {
+                                    finalResult.complete(consumeAuthorized);
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Namespace [{}] Role [{}] exception occurred while trying to check "
+                                                + "Consume permissions. {}", namespaceName, role, ex.getMessage());
+                                    }
+                                    finalResult.completeExceptionally(e);
```

Comment:
@yuruguo 

Something strange here, do we need ``e`` or ``ex`` ?
Besides, we need ``getCause()`` in the exception of CompletableFuture, otherwise, we will get CompletionException

Reply:
we need `ex` in finalResult.completeExceptionally()